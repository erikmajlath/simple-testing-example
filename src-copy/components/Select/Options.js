import React from 'react'

export default ({ options, value, onChange }) =>
  <ul>
    {options.map(item => {
      const selected = value.includes(item.value)
      return (
        <li key={item.value} onClick={() => onChange(item, selected)}>
          {item.label}
          <span>
            {selected && ' - SELECTED!'}
          </span>
        </li>
      )
    })}
  </ul>

// Constants
export const SF = { value: 'sf', label: 'San Francisco' }
export const LA = { value: 'la', label: 'Los Angeles' }
export const NY = { value: 'ny', label: 'New York City' }

export const OPTIONS = [SF, LA, NY]
