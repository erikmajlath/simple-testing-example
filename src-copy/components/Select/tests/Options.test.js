import React from 'react'
import { shallow } from 'enzyme'
import Options, { SF, LA } from '../Options'

const plus = (x, y) => x + y

describe('Options', () => {
  it('should test plus func', () => {
    expect(plus(3, 4)).toBe(7)
  })

  it('should render items', () => {
    const wrapper = shallow(<Options options={[SF, LA]} value={[]} />)

    expect(wrapper.find('li').length).toBe(2)
  })

  it('should render selected items', () => {
    const wrapper = shallow(<Options options={[SF, LA]} value={['la']} />)

    expect(wrapper.find({ children: ' - SELECTED!' }).length).toBe(1)
  })

  it('should match snapshot', () => {
    const wrapper = shallow(
      <Options options={[SF, LA]} value={[]} onChange={() => true} />
    )

    expect(wrapper).toMatchSnapshot()
  })
})
