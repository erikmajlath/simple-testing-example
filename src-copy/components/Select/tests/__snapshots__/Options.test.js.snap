// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options should match snapshot 1`] = `
<ul>
  <li
    onClick={[Function]}
  >
    San Francisco
    <span />
  </li>
  <li
    onClick={[Function]}
  >
    Los Angeles
    <span />
  </li>
</ul>
`;
